tests_e2e/orchestrator/lib/agent_test_suite.py

```diff
@@ -298,113 +308,136 @@ def _setup_node(self) -> None:
-            self._log.info("Collecting logs on test machine [%s]...", self.context.node.name)
+            self.context.lisa_log.info("Collecting logs on test node")
+            log.info("Collecting logs on test node")
             stdout = self.context.ssh_client.run_command("collect-logs", use_sudo=True)
-            self._log.info(stdout)
+            log.info(stdout)
 
             # Copy the tarball to the local logs directory
             remote_path = "/tmp/waagent-logs.tgz"
-            local_path = self.context.log_path/'{0}.tgz'.format(self.context.image_name)
-            self._log.info("Copying %s:%s to %s", self.context.node.name, remote_path, local_path)
+            local_path = self.context.log_path/'{0}.tgz'.format(self.context.environment_name)
+            log.info("Copying %s:%s to %s", self.context.node.name, remote_path, local_path)
             self.context.ssh_client.copy_from_node(remote_path, local_path)
 
         except:  # pylint: disable=bare-except
-            self._log.exception("Failed to collect logs from the test machine")
+            log.exception("Failed to collect logs from the test machine")
 
-    @TestCaseMetadata(description="", priority=0)
-    def agent_test_suite(self, node: Node, environment: Environment, variables: Dict[str, Any], log_path: str, log: Logger) -> None:
+    # NOTES:
+    #
+    #    * environment_status=EnvironmentStatus.Deployed skips most of LISA's initialization of the test node, which is not needed
+    #      for agent tests.
+    #
+    #    * We need to take the LISA Logger using a parameter named 'log'; this parameter hides tests_e2e.tests.lib.logging.log.
+    #      Be aware then, that within this method 'log' refers to the LISA log, and elsewhere it refers to tests_e2e.tests.lib.logging.log.
+    #
+    # W0621: Redefining name 'log' from outer scope (line 53) (redefined-outer-name)
+    @TestCaseMetadata(description="", priority=0, requirement=simple_requirement(environment_status=EnvironmentStatus.Deployed))
+    def main(self, node: Node, environment: Environment, variables: Dict[str, Any], working_path: str, log_path: str, log: Logger):  # pylint: disable=redefined-outer-name
+        """
+        Entry point from LISA
+        """
+        self._initialize(node, variables, working_path, log_path, log)
+        self._execute(environment, variables)
+
+    def _execute(self, environment: Environment, variables: Dict[str, Any]):
         """
         Executes each of the AgentTests included in the "c_test_suites" variable (which is generated by the AgentTestSuitesCombinator).
         """
-        self._set_context(node, variables, log_path, log)
+        # Set the thread name to the name of the environment. The thread name is added to each item in LISA's log.
+        with _set_thread_name(self.context.environment_name):
+            log_path: Path = self.context.log_path/f"env-{self.context.environment_name}.log"
+            with set_current_thread_log(log_path):
```

Comment:
each environment now has its own log

Reply:
sorry I'm really confused here, how many different sets of logs we have? 

For example
```

        self.context.lisa_log.info("Setting up test node")
        log.info("")
```
The first log goes to main log which is gist of each step of the flow or only logs specific msgs?
The second is what you are calling environment log which has something like node step, installing test version logs etc..?

where about test scenario log goes? will it have separate log?

Reply:
@nagworld9 I'd recommend checking the artifacts in the test run to see each log and its contents

Reply:
makes sense. 
Basically log.info could be either test log or environment log depending on which thread under its being invoked.

Reply:
Not quite.

We have multiple threads executing multiple environments, each environment executing multiple test suites. For each environment, we have an environment log and multiple test suite logs. If we had 3 environments executing 5 test suites each, we would have 3 environment logs + 3x5 test suite logs (total 18 logs, plus some other logs: the LISA log, the CheckAgentLog log, etc).

For each thread we have this nesting

```
log_path: Path = self.context.log_path/f"env-{self.context.environment_name}.log"
with set_current_thread_log(log_path):
    ....
    log.info("GOES TO THE ENVIRONMENT LOG")
    ...  
    for suite in self.context.test_suites:  # pylint: disable=E1133
    ....
        log_path:Path = self.context.log_path/f"{suite_full_name}.log"
        with set_current_thread_log(log_path):
             ....
            log.info("GOES TO THE TEST SUITE LOG")
    ....
    log.info("GOES TO THE ENVIRONMENT LOG AGAIN")

```

Reply:
i.e the scope of the log is determined by the context manager in the with statement